internal/driver/glfw/canvas.go

```diff
@@ -386,6 +389,21 @@ func (c *glCanvas) objectTrees() []fyne.CanvasObject {
 	return trees
 }
 
+func (c *glCanvas) overlayChanged(prevFocusMgr *app.FocusManager) {
+	c.setDirty(true)
+	c.RLock()
+	if prevFocusMgr == nil {
+		prevFocusMgr = c.focusMgr
+	}
+	focusMgr := c.overlays.TopFocusManager()
+	if focusMgr == nil {
+		focusMgr = c.focusMgr
+	}
+	c.RUnlock()
+	prevFocusMgr.FocusLost()
+	focusMgr.FocusGained()
```

Comment:
Is it possible for `prevFocusMgr == focusMgr`?

Reply:
Good catch! It might be possible with concurrent changes to the overlay stack.
I’ll add missing locking to the stack and also change the callback to be safe by design.

Reply:
Instead of locking, what if you just return early if they are equal?